components/gitpod-protocol/package.json

```diff
@@ -24,19 +24,24 @@
         "@types/random-number-csprng": "^1.0.0",
         "@types/uuid": "^8.3.1",
         "@types/ws": "^5.1.2",
+        "@typescript-eslint/eslint-plugin": "^4.5.0",
+        "@typescript-eslint/parser": "^4.5.0",
         "chai": "^4.3.4",
         "chai-subset": "^1.6.0",
+        "eslint": "^7.11.0",
         "mocha": "^5.0.0",
         "rimraf": "^2.6.2",
         "ts-node": "^10.4.0",
-        "typescript": "~4.4.2",
+        "typescript": "^4.4.4",
         "typescript-json-schema": "^0.50.1",
         "typescript-parser": "^2.6.1"
     },
     "scripts": {
         "prepare": "yarn run clean && yarn run build",
         "clean": "rimraf lib",
-        "build": "tsc",
+        "build": "yarn lint && tsc",
+        "lint": "yarn eslint src/**/*.ts",
+        "format": "yarn eslint src/**/*.ts --fix",
         "test": "mocha --opts mocha.opts './**/*.spec.ts' --exclude './node_modules/**'",
         "test-debug": "mocha --opts mocha.opts --inspect-brk './**/*.spec.ts' --exclude './node_modules/**'",
         "watch": "leeway exec --package .:lib --transitive-dependencies --filter-type yarn --components --parallel -- tsc -w --preserveWatchOutput"
```

Comment:
why do we use leeway for watching?

Reply:
It allows to get the list of all dependencies, and starts a watch in all of them.

Reply:
Just ran that command and it froze my workspace 😬

Reply:
Eslint doesn't have a watch mode, so we would need to use an extra package. vscode does however check the lining rules while editing. is that not good enough for your use case?
In any case, it would be great if we can land this without watch mode.

Reply:
> Eslint doesn't have a watch mode

The watch is handled by leeway. Yes, I can make a follow-up.